Add pull request template (#411)
virtool/virtool-ui: .github/pull_request_template.md

> ## Changes:

Reviewer: ```suggestion
## Changes
```

<!--- Provide the broad strokes of the changes made in a bulleted list --->

-

> ## Compatiblity:

Reviewer: ```suggestion
## Compatiblity
```

Any changes that require a newer version of the API than the oldest currently 

> supported version are considered breaking changes.

Reviewer: By "oldest current", do you mean "most recent" or "last"?

Common breaking changes:

>  -  Using a new API not available in oldest currently supported server
>  -  Requiring new elements in an existing response from the API

Reviewer: ```suggestion
 -  Making a request to a new API path that will return `404` in older versions of the API.
 -  Requiring new fields from an API endpoint that will not be present in responses from older versions of the API.
```

>  -  Dropping support for a previously valid response model from the API.

Reviewer: It took me a while to figure out where you were getting at with this. The other ones are obvious, but I think this needs a brief example.

Choose one:

>  - [ ] The changes maintain compatibility with the existing API

Reviewer: ```suggestion
 - [ ] The changes do not break compatibility
```

>  - [ ] The changes introduce a potential breaking change

Reviewer: ```suggestion
 - [ ] The changes potentially break compatibility
```

> ## Pre-Review Checklist:

Reviewer: ```suggestion
## Pre-Review Checklist
```

 - [ ] All changes are tested
 - [ ] All touched code documentation is updated
 - [ ] All tests pass in your local environment

>  - [ ] Deepsource issues have been reviewed and resolved

Reviewer: Should leave this open to ignoring bad issues.

 - [ ] Debugging logging and commented out code has been removed

 

> ## Screenshots (for UI changes):
> <!--- Provide screenshots of all visual changes made in the PR --->

Reviewer: ```suggestion
## Screenshots

_Only required for visual changes_.
```

<!---

    Please initially open your PR as a draft until CI has passed and the

>     PR is ready for review. 

Reviewer: I am not a fan of this policy. GitHub shows whether CI has passed. I don't think we need a manual indicator of this.

What happens in practice is that people open PRs, CI passes, and then they forget to de-draft.

>     Once marked as ready for review, avoid force-pushing to the branch. 

Reviewer: When would they not be able to avoid a force push? They should NEVER do it.

Author: Removed comment about that all-together. imo probably should be part of training rather than the template anyway

--->

## PR type: 
<!---   
PR types: (check all applicable)
        Breaking change: Any change that either drops support for a 
        preview server version
        Bug fix: Restores functionality of a currently broken feature
        Build: Makes changes to how the image is built. Ex: update Vite 
        version, changing the dockerfile
        CI: Changes to our CI configuration files and scripts
        Documentation: Updates in-code or external documentation
        Feature: Introduce a new feature or rework an existing one
        Optimization: Improves performance
        Refactor: Improves code quality without changing the user experience

> --->

Reviewer: I don't see the value of this.

The PR should already be dealing with a Linear issue, which has the appropriate labelling. Otherwise, GitHub labels should be used.

- [ ] Breaking change
- [ ] Bug fix 
- [ ] Build
- [ ] CI
- [ ] Documentation
- [ ] Feature
- [ ] Optimization
- [ ] Refactor
- [ ] Other - Please specify

## Changes:
<!--- Provide the broad strokes of the changes made in a bulleted list --->

-

## Screenshots (for UI changes):

> <!--- Provide screenshots of all visual changes made in the PR --->

Reviewer: Put this last so that it is not necessary to scroll through the images to get to the last text item.

## Added/Updated tests?
<!--- Ensure that all touched components are tested for functionality --->

- [ ] Yes
- [ ] Yes, comment:
- [ ] No, why:

> 

Reviewer: * Not just components.
* I think this should just be one checkbox. When are they going to not write tests?
